docs/desktop-integration.rst

```diff
@@ -144,48 +144,121 @@ it will work as expected from within a sandboxed application.
+As of Flatpak 0.10.1, Flatpak can detect the active Gtk theme on host by
+reading the value of the ``gtk-theme`` DConf key. If the corressponding
+theme is packaged as an extension, Flatpak will automatically install it
+during ``flatpak install`` or ``flatpak update``.
 
-  $ flatpak install kdeapps org.kde.KStyle.Adwaita//5.9
-  $ flatpak install kdeapps org.kde.PlatformTheme.QGnomePlatform//5.9
+Instructions for Qt themes
+``````````````````````````
+
+There are a few Qt theme extensions packaged on Flathub. To see a list,
+you can run::
+
+ $ flatpak remote-ls --columns=application flathub | grep org.kde.KStyle
+
+Then you can install the theme with::
+
+ $ flatpak install flathub org.kde.KStyle.Foo
+
+replacing ``Foo`` with the name of the desired theme. The theme needs
+to be available for the KDE runtime branch used by the application.
+
+Gtk look and feel in Qt applications
+````````````````````````````````````
+
+Since ``org.kde.Platform`` runtime branches 5.15-22.08+ and 6.5+, on
+Wayland, ``org.kde.WaylandDecoration.QAdwaitaDecorations`` and
+``org.kde.WaylandDecoration.QGnomePlatform-decoration`` needs to be
+installed. Please see the `usage <https://github.com/FedoraQt/QAdwaitaDecorations?tab=readme-ov-file#usage>`_
+instructions upstream too.
+
+These plugins will be part of upstream starting at Qt 6.8.
+
+For older runtimes, ``org.kde.PlatformTheme.QGnomePlatform`` and
+``org.kde.WaylandDecoration.QGnomePlatform-decoration`` needs to be
+installed.
 
 Applying themes
 ^^^^^^^^^^^^^^^
 
-There is no ideal way to specify the theme Flatpak applications use. The applications will try to match the system theme currently being used, if it corresponds to any of the Flatpak themes installed, and will fall back to Adwaita (if they use Gtk2 or Gtk3) or the default Qt theme (if they use Qt) if a corresponding theme isn't detected.
+The pre-requisite to applying themes in Flatpaks is to have the theme
+available to the application in the sandbox, commonly done by packaging
+them as theme extensions.
+
+Now the system theme needs to be communicated from host to the sandbox.
+
+On X11, Gtk3 picks up themes from XSettings. Specifically, on GNOME, the
+GNOME XSettings daemon ``gsd-xsettings`` reads the DConf values and
+converts them into XSettings values. On GNOME, this should work by default
+provided ``gsd-xsettings`` is running.
+
+On Wayland, the active theme is exposed via the corresponding Settings
+portal. This requires the portal backend for the desktop environment to
+be installed.
+
+Once the theme is installed and exposed in the sandbox, depending on the
+toolkit/libraries used or the application, it will automatically be
+applied.
+
+If the theme is not available via Flatpak extensions or portal
+or xsettings support is lacking, Gtk and Qt applications will fallback
+to using Adwaita or the default Qt theme.
+
+In this case, either the theme needs to be packaged as an extension
```

Comment:
It isn't pretty but the command on my blog post creates a package of the users theme:

```bash
#!/bin/bash
DEFAULT_ARCH=`flatpak --default-arch`
THEME_NAME=`gsettings get org.gnome.desktop.interface gtk-theme | tr -d \'`
THEME_EXTENSION_DIR=~/.local/share/flatpak/extension/org.gtk.Gtk3theme.$THEME_NAME/$DEFAULT_ARCH/3.22

mkdir -p $THEME_EXTENSION_DIR

cp -r /usr/share/themes/$THEME_NAME/gtk-3.0/* $THEME_EXTENSION_DIR
```

Reply:
Slightly modified it and added it in the extensions page 3327a5165b0eff3c06233b21f2cfdb3f9c19ec83

Reply:
Nice improvement, thanks.

Reply:
Forgot to add the `/themes` to `XDG_DATA_HOME`